Dynamic backend generation is not working.

**Describe the bug**
Dynamic backend generation is not working

**To Reproduce**
Steps to reproduce the behavior:
1. use a large vcl template (>1MB , provided as volume)
2. respawn/recreate multiple (example 4 backends at once
3. old PODs will not be removed from backend
4. new PODs will not be added to backend


**Expected behavior**
kube-httpcache adds/removes backends


**Environment:**
 - Kubernetes version: 1.18
 - kube-httpcache version: [latest master]

**Logs**

```
panic: runtime error: index out of range [0] with length 0

goroutine 8 [running]:
github.com/mittwald/kube-httpcache/pkg/watcher.(*EndpointWatcher).watch(0xc0002e5a90, 0xc00004a480, 0xc00004a4e0)
	/workspace/pkg/watcher/endpoints_watch.go:93 +0xf1d
created by github.com/mittwald/kube-httpcache/pkg/watcher.(*EndpointWatcher).Run
	/workspace/pkg/watcher/endpoints_watch.go:17 +0x98

```

Or

```
W0219 16:27:58.037023       1 run.go:53] error while watching for updates: error while loading VCL (code 106): Message from C-compiler:
gcc: internal compiler error: Killed (program cc1)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-6/README.Bugs> for instructions.
Running C-compiler failed, exited with 4
VCL compilation failed
```

--- a/pkg/watcher/endpoints_watch.go
+++ b/pkg/watcher/endpoints_watch.go
@@ -77,6 +77,13 @@ func (v *EndpointWatcher) watch(updates chan *EndpointConfig, errors chan error)
 
 				addresses = append(addresses, a)
 			}
+
+			if len(addresses) == 0 {
+				glog.Warningf("service '%s' has no endpoint that is ready", v.serviceName)
+				v.endpointConfig = NewEndpointConfig()
+				continue
+			}
+
 			endpoint.Subsets[0].Addresses = addresses
 
 			newConfig := NewEndpointConfig()

